internal/userutils/userutils.go

```diff
@@ -0,0 +1,40 @@
+// Package userutils provides functions related to system users and groups.
+package userutils
+
+import (
+	"errors"
+	"fmt"
+	"os"
+)
+
+// GroupFile is the path to the group file. It is exported for testing purposes.
+var GroupFile = "/etc/group"
+
+// LockGroupFile creates a lock file at /etc/group.lock, the same location
+// used by tools like gpasswd and groupmod to prevent concurrent modifications
+// to the /etc/group file.
+// The lock file contains the PID of the process that created it.
+func LockGroupFile() (err error) {
+	lockFile, err := os.OpenFile(GroupFile+".lock", os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0600)
```

Comment:
I see the [original implementation](https://salsa.debian.org/debian/libuser/blob/7abf20577065bf0d48703ece1c0a1b18a4bc2e67/modules/files.c#L291-L291) is doing things a bit differently:
 - Creates a temporary file
 - Tries to hard-link it

gpasswd does something similar but using a `/etc/group.pid` as source file, but that seems to me even unsafer.

Maybe because it can be a faster operation... However what we do here with `os.EXCL` and with sync, I think it should be enough too.

The only thing is that if an app is fast enough to check if the file exists, then it may return the `Invalid contents of lock `%s'` error instead of telling us what it is, but it shouldn't matter much.

```suggestion
	lockFile, err := os.OpenFile(GroupFile+".lock", os.O_CREATE|os.O_EXCL|os.O_WRONLY|os.O_SYNC, 0600)
```